### DISCUSSION TITLE
Video 281 - Creating an EffNetB2 Feature Extractor Model - invalid hash

### DISCUSSION THREAD

**Discussion by MariaKrakowiak** [mrdbourke/pytorch-deep-learning#693] [Q&A] [answered]
Hi All,

I have come across a problem with below fragment of code:

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/59764846/0577e109-24cf-4241-ace9-489d10dc0ef2)

I have received below:

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/59764846/6778f757-a397-4f66-a8fd-c80d30638254)

Did anyone have similar troubles? If so - could you please advise?

Thanks in advance :)

**Comment by gulnuravci** [comment, marked as answer]
Hi @MariaKrakowiak, I just had the same issue. Here is the fix I found that worked for me: [https://github.com/pytorch/vision/issues/7744.](https://github.com/pytorch/vision/issues/7744.) Hope this helps!

**Reply by gulnuravci** [reply]
```python
from torchvision.models import efficientnet_b0, EfficientNet_B0_Weights
from torchvision.models._api import WeightsEnum
from torch.hub import load_state_dict_from_url

def get_state_dict(self, *args, **kwargs):
    kwargs.pop("check_hash")
    return load_state_dict_from_url(self.url, *args, **kwargs)
WeightsEnum.get_state_dict = get_state_dict

efficientnet_b0(weights=EfficientNet_B0_Weights.IMAGENET1K_V1)
efficientnet_b0(weights="DEFAULT")
```

**Reply by gulnuravci** [reply]
Specifically this ^

**Reply by MariaKrakowiak** [reply]
Hi @gulnuravci - your suggestion works. Many thanks for your advice - much appreciated 👍

**Reply by mrdbourke** [reply]
Hi @gulnuravci , 

Thank you for the solution! I didn't know about this.

Also for those wondering, see the link to the solution comment here: https://github.com/pytorch/vision/issues/7744#issuecomment-1757321451

From the looks of that thread^ it seems like a PyTorch `torchvision` issue that should be fixed in a future version.

**Reply by gulnuravci** [reply]
@MariaKrakowiak I'm glad to hear that and @mrdbourke I'm glad to have helped, thank you for all your efforts, it has been inspiring!
